pkg/app/registry/static/static.go

```diff
@@ -148,23 +148,34 @@ func (b *reg) ListProviders(ctx context.Context) ([]*registrypb.ProviderInfo, er
 	return providers, nil
 }
 
-func (b *reg) ListSupportedMimeTypes(ctx context.Context) (map[string]*registrypb.AppProviderList, error) {
+func (b *reg) ListSupportedMimeTypes(ctx context.Context) ([]*registrypb.MimeTypeInfo, error) {
 	b.RLock()
 	defer b.RUnlock()
 
-	mimeTypes := make(map[string]*registrypb.AppProviderList)
+	res := []*registrypb.MimeTypeInfo{}
+	mtmap := make(map[string]*registrypb.MimeTypeInfo)
 	for _, p := range b.providers {
 		t := *p
 		t.MimeTypes = nil
 		for _, m := range p.MimeTypes {
-			if _, ok := mimeTypes[m]; ok {
-				mimeTypes[m].AppProviders = append(mimeTypes[m].AppProviders, &t)
+			if _, ok := mtmap[m]; ok {
+				mtmap[m].AppProviders = append(mtmap[m].AppProviders, &t)
 			} else {
-				mimeTypes[m] = &registrypb.AppProviderList{AppProviders: []*registrypb.ProviderInfo{&t}}
+				mtmap[m] = &registrypb.MimeTypeInfo{
+					MimeType:     m,
+					AppProviders: []*registrypb.ProviderInfo{&t},
+					Ext:          "", // TODO fetch from config
+					Name:         "",
+					Description:  "",
+					Icon:         "",
+				}
 			}
 		}
 	}
-	return mimeTypes, nil
+	for _, m := range mtmap {
```

Comment:
Why add the cached elements to the result list at the end?
Just do it when creating the new MimeTypeInfo

Reply:
That's right, I had changed approach following discussions with @ishank011 and trusting you guys as Go experts ;-) but as a matter of fact `mtmap` is a cache of pointers, so once allocated they can be pushed to the result even if the structs they point to have their `AppProviders` lists being modified.